test/integration/rpc_test.lua

```diff
@@ -119,6 +119,13 @@ function g.test_routing()
     t.assert_equals(res.uuid, B2.instance_uuid)
     t.assert_equals(res.peer, B2.net_box:call('box.session.peer'))
 
+    local res, err = rpc_call(B2,
+        'myrole', 'get_session', nil,
+        {prefer_local = false}
+    )
+    t.assert_not(err)
+    t.assert_not_equals(res.peer, B2.net_box:call('box.session.peer'))
```

Comment:
The test doesn't always reproduce an error even without a fix. The `rpc_call` may be performed on B1, and, of course, peer would differ in this case.